src/components/SharePanel/SharePanel.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react'
+import {getStoredState} from '../../utils/statestore'
+import {throttle} from 'lodash'
+const styles: any = require('./SharingPanel.module.styl')
+
+interface Props {
+}
+
+interface State {
+  display: boolean
+}
+
+type Size = 'large' | 'medium' | 'small'
+const millUntilDisplay = 3 * 60 * 1000
+
+export default class SharePanel extends React.Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      display: false,
+    }
+
+    this.onResize = throttle(this.onResize.bind(this), 100)
+  }
+
+  componentDidMount() {
+    const lastTimestamp = getStoredState().initialLoadTimestamp
+    const currentTimestamp = Date.now()
+
+    if ((currentTimestamp - lastTimestamp) < millUntilDisplay) {
+      setTimeout(() => {
+        this.setState({display: true} as State)
+      }, (millUntilDisplay - (currentTimestamp - lastTimestamp)))
+    } else {
+      this.setState({display: true} as State)
+    }
+
+    window.addEventListener('resize', this.onResize, false)
+  }
+
+  onResize(e) {
+    this.forceUpdate()
+  }
+
+  render() {
+
+    const displayMode: Size = ((w: number): Size => {
+      if (w <= 1600) {
+        return 'small' as Size
+      } else if (w <= 2000) {
+        return 'medium' as Size
+      } else {
+        return 'large' as Size
+      }
+    })(window.innerWidth)
+
+    const shareTitle = 'I just learned how to develop apps with Apollo and GraphQL'
```

Comment:
given that this panel is used during the exercises, we could use another share title. Something like "Currently learning Apollo and GraphQL"?